components/director/pkg/webhook/data_input_builder.go

```diff
@@ -0,0 +1,173 @@
+package webhook
```

Comment:
I was wondering where to put it but the `internal/domain` didn't seem to be fitting, which directory do you refer to in `internal/domain` - `webhook` or on root level - `internal/domain`?
I used the pkg because the other structures that are used in this file are in `pkg/webhook` but agree it's not a very good place.